### ISSUE TITLE
Global macros should be possible to store in a file even on Windows

### ISSUE THREAD

**Issue by xdy** [RPTools/maptool#1449]
**Is your feature request related to a problem? Please describe.**
On windows Global Macros are stored in the registry, which has a limited size, which sometimes causes problems.

**Describe the solution you'd like**
I'd like to be able to store macros in files, just like other configuration stuff.

**Describe alternatives you've considered**
Telling people to not store big macros in Global Macros.

**Additional context**
Comes from a Discord discussion in #javascript-frame5-dialog5

The following Stackoverflow will probably be useful when implementing.
https://stackoverflow.com/questions/208231/is-there-a-way-to-use-java-util-preferences-under-windows-without-it-using-the-r/208289#208289

It may or may not be a good idea to allow switching to any implemented backend, on all OS:s.

**Comment by Azhrei** [comment]
Yes.  Probably `$MAPTOOL_DATADIR/macros` would make sense.

After seeing this in Discord, I was wondering if there's any benefit to having subdirectories under it, but I can't think of any reason to do so.

On systems that use files currently, they are typically named with a long string of digits; maybe we could use the name of the macro (sanitized for the file system, of course)...

**Comment by Phergus** [comment]
Macros do have IDs so that could be used even though it wouldn't be immediately identifiable to a user.  Use macro names can be a little complicated just because people to like to put images instead of names.  You an also have multiple macros with the same name in different groups or the same group for that matter.

**Comment by cwisniew** [comment]
> Macros do have IDs so that could be used even though it wouldn't be immediately identifiable to a user. Use macro names can be a little complicated just because people to like to put images instead of names. You an also have multiple macros with the same name in different groups or the same group for that matter.

Yeah I agree ID's are probably the way to go, they should be unique, you don't have to worry about deleting files when someone changes the name of the macro.
